### PR TITLE
Add Asypadding for conv fusion.

### PR DESCRIPTION
**Pull request by zhaoyuchen2018**
test=develop

reference: pr/20042

Signed-off-by: zhaoyuchen <zhaoyuchen01@baidu.com>

### CODE REVIEW

File: paddle/fluid/operators/fused/conv_fusion_op.cu [hunk 1]
```diff
@@ -33,6 +36,44 @@ using framework::AlgorithmsCache;
 template <typename T>
 using ScalingParamType = typename platform::CudnnDataType<T>::ScalingParamType;
 
+static inline bool IsSymmetricPadding(const std::vector<int>& paddings,
```
Comment (liym27) [inline paddle/fluid/operators/fused/conv_fusion_op.cu]: `IsSymmetricPadding` and `PadFunction` have been defined in  file **padding.h**,  better to  call them in [**padding.h**](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/math/padding.h).

File: paddle/fluid/operators/fused/conv_fusion_op.cc [hunk 2]
```diff
@@ -90,7 +111,8 @@ class Conv2DFusionOpInferShape : public framework::InferShapeBase {
       std::vector<framework::DDim> oshapes;
       oshapes.reserve(channels.size());
       for (size_t i = 0; i < channels.size(); ++i) {
-        oshapes.push_back({oshape[0], channels[i], oshape[2], oshape[3]});
+        oshapes.push_back(
+            {output_shape[0], channels[i], output_shape[2], output_shape[3]});
       }
       ctx->SetOutputsDim("Outputs", oshapes);
```
Comment (liym27) [inline paddle/fluid/operators/fused/conv_fusion_op.cc:166]: The computational logic of the above code is the same as [`ConvOp::InferShape`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/conv_op.cc#L33), and there are more checks on parameters in `ConvOp::InferShape` .
Better to use `ConvOp::InferShape`.

File: paddle/fluid/operators/fused/conv_fusion_op.cu (hunk 1 above)
Comment (zhaoyuchen2018) [reply paddle/fluid/operators/fused/conv_fusion_op.cu]: done

File: paddle/fluid/operators/fused/conv_fusion_op.cc (hunk 2 above)
Comment (zhaoyuchen2018) [reply paddle/fluid/operators/fused/conv_fusion_op.cc:166]: when compute split channels need output dims, cannot call conv::infershape directly.

**Comment by liym27** [comment]
LGTM

**Review by wangchaochaohu — APPROVED** [review]
LGTM

**Pull request merged**